community/ndp/transport-socket/src/main/java/org/neo4j/ndp/transport/socket/SocketProtocolV1.java

```diff
@@ -70,7 +71,7 @@ public SocketProtocolV1( final Log log, Session session )
         this.session = session;
         this.output = new ChunkedOutput();
         this.input = new ChunkedInput();
-        this.packer = new PackStreamMessageFormatV1.Writer( output, output.messageBoundaryHook() );
+        this.packer = new PackStreamMessageFormatV1.Writer( new PackStream.Packer( output ), output.messageBoundaryHook() );
```

Comment:
I kind of agree with both here - it's the "right thing" to do, to not new up objects inside constructors. However, we also don't want to be designing the java io streams API.

The PackStreamMessageFormatV1 is strictly tied to use a PackStream packer, while it's comp-sci-wrong to have two constructors here, I think it's pragmatically-right to do it this way. 

I'll make that modification and then merge this in.